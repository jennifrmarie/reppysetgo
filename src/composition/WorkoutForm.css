* {
    color: white;
}
.workout-box {
    background-color: #ece8e8fb;
    background-image: url('https://i.imgur.com/1qCGEVN.png');
    background-size: contain;
    background-repeat: no-repeat;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    position: relative;
}

.logo__nav {
    position: relative;
    left: 200px;
}

.NotePage__back-button {
    width: 150px;
    background-color: black;
    position: absolute;
    left: -10px;
    padding: 10px;
    font-size: 20px;
}
.workout-form {
    background-color: black;
    justify-content: space-evenly;
    border: 2px solid white;
    position: relative;
    top: 100px;
    left: -90px;
    width: 23.5%;
    height: 100%;
    padding-top: 20px;
    min-width: 150px;
    text-align: center;
    font-size: 20px;
    font-weight: bold;
    letter-spacing: 5px;
}

.form-box1, .form-box2, .form-box3, .form-box4 {
    clear: both;
    margin: 10px;
    height: 40px;
    max-height: 20px;
    border: 5px solid black;
    color: black;
    font-size: 20px;
    max-width: 160px;
    margin-right: 10px;
}

.form-box2, .form-box3, .form-box4 {
    width: 60px;
    /* margin-left: 120px; */
    align-self: center;

}

.form-box1 {
    width: 150px;
    min-width: 50px
}

.workout-name-input {
    margin: 10px;
}

.submit-button {
    position: relative;
    background-color: black;
    color: white;
    margin-top: 20px;
    top: -10px;
    width: 100px;
    font-size: 20px;
    padding: 10px;
}

.submit-button:hover {
    cursor: pointer;
    background-color: white;
    color: black;
}

.app__title {
    color: black;
    margin-left: 60px;
    letter-spacing: .5em;
}

.list_page {
    position: relative;
    top: 500px;
}

@media only screen and (max-width: 700px) {
    .workout-form {
        top: 50px;
        font-size: 20px;
        width: 150px;
        position: relative;
        left: 100px;
        margin-right: 80px;
        font-size: 15px;
    }

    .workout-box {
        background-size: 50%;
        display: flex;
        flex-direction: column;
    }
    .NotePage__back-button {
        width: 80px;
        border-radius: .5em;
    }

    .form-box1, .form-box2, .form-box3, .form-box4 {
        font-size: 15px;
    }
    .form-box1 {
        width: 80px;
    }
}

@media only screen and (max-width: 1000px) and (min-width: 700px) {
    .workout-box {
        display: flex;
        flex-direction: column;
    }

    .workout-form {
        display: flex;
        flex-direction: column;
       position: relative;
       left: -30px;
       width: 300px;
    }
    .submit-button {
        left: 110px;
    }
}