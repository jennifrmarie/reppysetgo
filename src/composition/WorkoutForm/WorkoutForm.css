* {
    color: white;
}
.workout-box {
    background-color: #ece8e8fb;
    background-image: url('https://i.imgur.com/1qCGEVN.png');
    background-size: contain;
    background-repeat: no-repeat;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.logo__nav {
    position: relative;
    left: 200px;
}

.NotePage__back-button {
    width: 150px;
    background-color: black;
    position: absolute;
    left: -10px;
    padding: 10px;
    font-size: 20px;
}

.NotePage__back-button:hover {
    cursor: pointer;
    background-color: #ad4141fb;;
}


.workout-form {
    padding: 30px;
    width: 30%;
    background-color: black;
    justify-content: space-evenly;
    border: 2px solid white;
    /* margin-top: 50px; */
    line-height: 2em;
    flex-direction: column;
    display: flex;
    height: 100%;
    text-align: center;
    font-size: 30px;
    font-weight: bold;
    letter-spacing: 5px;
}

.form-box1, .form-box2, .form-box3, .form-box4 {
    border: 5px solid black;
    color: black;
    font-size: 20px;
    float: right;
    line-height: 2em;
}

.form-box2, .form-box3, .form-box4 {
    width: 60px;
    align-self: center;
}

.form-box1 {
    width: 150px;
    min-width: 50px;
}


.workout-name-input {
    margin-left: 70px;
}

.submit-button {
    display: flex;
    align-self: center;
    justify-self: center;
    background-color: black;
    color: white;
    margin-bottom: 5px;
    margin-top: 5px;
    font-size: 25px;
    padding: 10px;
}

.submit-button:hover {
    cursor: pointer;
    background-color: #ad4141fb;
    color: white;
}

.app__title {
    color: black;
    margin-left: 60px;
    letter-spacing: .5em;
}

.list_page {
    position: relative;
    top: 500px;
}

@media only screen and (max-width: 700px) {
    .workout-form {
        font-size: 20px;
        font-size: 15px;
        width: 50%;
        padding: 10px 10px 10px 0px;
    }

    .NotePage__back-button {
        width: 100px;
        border-radius: .3em;
    }

    .form-box1 {
        width: 120px;
    }

    .workout-name-input {
        margin-left: 50px;
    }

    .submit-button {
        font-size: 15px;
        margin-top: 10px;
    }

    .workout-box {
        
        background-size: 50%;
        display: flex;
        flex-direction: column;
    }

    .form-box1, .form-box2, .form-box3, .form-box4 {
        font-size: 15px;
        line-height: 1em;
    }
}

@media only screen and (max-width: 1000px) and (min-width: 700px) {
    .workout-box {
        display: flex;
        flex-direction: column;
        
    }
    .NotePage__back-button {
        width: 100px;

    }
    .form-box1, .form-box2, .form-box3, .form-box4 {
        line-height: 1.5em;
        font-size: 15px;
    }

    .form-box1 {
        width: 100px;
    }

    .workout-name-input {
        margin-left: 20px;
    }

    .workout-form {
        display: flex;
        flex-direction: column;
        justify-content: center;
        font-size: 20px;
        padding: 10px 10px 10px 0px;
    }
    .submit-button {
        margin-top: 10px;
        font-size: 15px;
    }
}