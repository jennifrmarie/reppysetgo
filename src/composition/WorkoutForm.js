import React, { Component } from 'react'
import AppContext from '../AppContext'
import uuid from 'uuid'
import WorkoutList from './WorkoutList'
import './WorkoutForm.css'
import NavButton from './NavButton'
import { withRouter } from 'react-router-dom'
import config from '../config'
import moment from 'moment';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 
class WorkoutForm extends Component {
    static defaultProps = {
        match: {
          params: {}
        },
        history: {
          push: () => { }
        }
      }
    static contextType = AppContext
    constructor(props) {
        super(props);
        this.state = {
            name: '',
            sets: '',
            reps: '',
            weight: '',
            date: '',
        };
        this.handleNameChange.bind(this);
        this.handleSubmit.bind(this)
        this.handleRepChange.bind(this)
        this.handleSetChange.bind(this)
        this.handleWeightChange.bind(this)
    }

    componentDidMount() {
        if(this.props.match.params.id) {
            const item = this.context.items.find(item => item.id === parseInt(this.props.match.params.id))
            this.setState({
                name: item.name,
                sets: item.sets,
                reps: item.reps,
                weight: item.weight
            })
        }
    }
    getAuthToken() {
        return window.localStorage.getItem(config.TOKEN_KEY)
      }    

    handleSubmit = (e) => {
        e.preventDefault()
        const data = {
            name: this.state.name,
            sets: this.state.sets,
            reps: this.state.reps,
            weight: this.state.weight,
            date: moment(this.props.match.params.dateId)
        }
  
        if(this.props.match.params.id) {
            const item = this.context.items.find(item => item.id === parseInt(this.props.match.params.id))
            item.name = this.state.name;
            item.sets = this.state.sets;
            item.reps = this.state.reps;
            item.weight = this.state.weight;
            this.context.editItem(item)

        } else {
            const item = {
                id: uuid(),
                name: this.state.name,
                sets: this.state.sets,
                reps: this.state.reps,
                weight: this.state.weight,
                date: moment(this.props.match.params.dateId,"MMDDYYYY")          
            }
            console.log(this.context.date)
            this.context.addItem(item)
           
        }
        
       
        this.setState({
            name: '',
            sets: '',
            reps: '',
            weight: '',
            date: '',
        })
      }
    
    handleNameChange(e) {
        this.setState({
            name: e.target.value,
        })
      }

    handleRepChange(e) {
        this.setState({
            reps: e.target.value
        })
    }

    handleSetChange(e) {
        this.setState({
            sets: e.target.value
        })
    }

    handleWeightChange(e) {
        this.setState({
            weight: e.target.value
        })
    }
    render() {
        const location = {
            pathname: '/dashboard',
            state: { dates: [] }
        }
        return (
            <div className="workout-box">
                <nav className="logo__nav"></nav>
                <form className="workout-form">
                    <div>
                    <label htmlFor='workout-name-input'>
                        LIFT:
                    </label>
                    <input 
                        className="form-box1"  
                        type="text" 
                        value={this.state.name}
                        placeholder="e.g. deadlift" 
                        required 
                        onChange={this.handleNameChange.bind(this)} 
                        name="Workout" />{' '} 
                    </div>
                    <div>
                    <label htmlFor='workout-set-input'>
                        SETS:
                    </label>
                    <input className="form-box2" 
                        type="text" align="right" 
                        value={this.state.sets}
                        placeholder="e.g. 3" required 
                        onChange={this.handleSetChange.bind(this)} 
                    />{' '}
                    </div>
                    <div>
                    <label htmlFor='workout-reps-input'>
                        REPS:
                    </label>
                    <input className="form-box3" 
                        type="text" align="right" 
                        value={this.state.reps}
                        placeholder="e.g. 5" 
                        required 
                        onChange={this.handleRepChange.bind(this)} 
                    />{' '}
                    </div>
                    <div>
                    <label htmlFor='workout-weight-input'>
                        WEIGHT:
                    </label>
                    <input className="form-box4" 
                        type="text" align="right" 
                        value={this.state.weight}
                        placeholder="e.g. 135" 
                        required 
                        onChange={this.handleWeightChange.bind(this)} 
                    />
                    </div>
                    <button 
                        tag='a' 
                        className="submit-button" 
                        onClick={this.handleSubmit}
                    >
                        Submit
                    </button>

                    </form>
                <WorkoutList
                    dateId={this.props.match.params.dateId}
                />
                <NavButton
                    tag='button'
                    role='link'
                    onClick={() => this.props.history.push(location)}
                    className='NotePage__back-button'
                    >
                    <FontAwesomeIcon icon='hand-point-left' size="lg" />
                </NavButton>
            </div>
        )
    }
}

export default withRouter(WorkoutForm)