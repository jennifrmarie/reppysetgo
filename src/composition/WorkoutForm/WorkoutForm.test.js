import React from 'react';
import ReactDOM from 'react-dom';
import WorkoutForm from './WorkoutForm';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<WorkoutForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<WorkoutForm name="Messages" unread={4}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });