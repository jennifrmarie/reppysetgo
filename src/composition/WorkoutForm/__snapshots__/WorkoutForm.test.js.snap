// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="workout-box"
>
  <nav
    className="logo__nav"
  />
  <form
    className="workout-form"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="workout-name-input"
      >
        LIFT:
      </label>
      <input
        className="form-box1"
        name="Workout"
        onChange={[Function]}
        placeholder="e.g. deadlift"
        required={true}
        type="text"
        value=""
      />
       
    </div>
    <div>
      <label
        htmlFor="workout-set-input"
      >
        SETS:
      </label>
      <input
        className="form-box2"
        inputMode="numeric"
        onChange={[Function]}
        pattern="[0-9]*"
        required={true}
        type="number"
        value=""
      />
       
    </div>
    <div>
      <label
        htmlFor="workout-reps-input"
      >
        REPS:
      </label>
      <input
        className="form-box3"
        inputMode="numeric"
        onChange={[Function]}
        pattern="[0-9]*"
        required={true}
        type="number"
        value=""
      />
       
    </div>
    <div>
      <label
        htmlFor="workout-weight-input"
      >
        WEIGHT:
      </label>
      <input
        className="form-box4"
        inputMode="numeric"
        onChange={[Function]}
        pattern="[0-9]*"
        required={true}
        type="number"
        value=""
      />
    </div>
    <button
      className="submit-button"
      tag="a"
    >
      Submit
    </button>
  </form>
  <ul
    className="workout-list"
  />
  <button
    className="NavButton NotePage__back-button"
    onClick={[Function]}
    role="link"
  />
</div>
`;
